### PR TITLE
Refactoring Processor

### PR DESCRIPTION
**Pull request by khemrajrathore**
This PR contains changes to help writing test case easily
- The change implies on frontend's Processor which extends BaseProcessor
- Earlier invoking `processCpg` on the processor yielded `Either[String, Unit]` which has now being changed to `Either[String, (Cpg, Map[String, Json])]`
- Left will have the error messages
- Right will have a tuple with 1st element being generated Cpg, 2nd element being the outputMap, which is dumped to privado.json)
-  Added a flag which doesn't close the cpg when exiting `processCpg` so that the cpg can be used to writing test cases
- Created Language specific test classes example for Java 
- For writing test cases similarly lang specific test classes can be created and used

### CODE REVIEW

**Comment by khemrajrathore** [comment]
1 test case is failing for TimeMetric as it is a global object

**Review by pandurangpatil — COMMENTED** [review]
one minor review comment rest looks good

**Comment by pandurangpatil** [comment]
> 1 test case is failing for TimeMetric as it is a global object

It shouldn't impact even if its a global object.

**Comment by khemrajrathore** [comment]
> > 1 test case is failing for TimeMetric as it is a global object
> 
> It shouldn't impact even if its a global object.

It is because, earlier TimeMetric was used in only 1 test case, but now when we run Processor in test case, it is invoked multiple times, leading to a conflict

**Comment by pandurangpatil** [comment]
> > > 1 test case is failing for TimeMetric as it is a global object
> > 
> > 
> > It shouldn't impact even if its a global object.
> 
> It is because, earlier TimeMetric was used in only 1 test case, but now when we run Processor in test case, it is invoked multiple times, leading to a conflict

Hmmm makes sense. Ignore the test case for now, as I intend to replace it with TimeMetric from Joern in the coming days. I will take care of this test case at that time. Add TODO: in comments though

File: src/main/scala/ai/privado/languageEngine/base/processor/BaseProcessor.scala [hunk 1]
```diff
@@ -51,24 +53,27 @@ abstract class BaseProcessor(
   /** Entry method to read files and generate output
     * @return
     */
-  def processCpg(): Either[String, Unit] = ???
+  def processCpg(): Either[String, (Cpg, Map[String, Json])] = ???
```
[the remark this answers is not in the document]
Comment (khemrajrathore) [reply src/main/scala/ai/privado/languageEngine/base/processor/BaseProcessor.scala]: Done

**Review by pandurangpatil — APPROVED** [review]

**Pull request merged**